crates/workspace/src/workspace.rs

```diff
@@ -3060,14 +3067,20 @@ impl Workspace {
                     if let Some(item) = pane.active_item() {
                         item.workspace_deactivated(cx);
                     }
-                    if matches!(
-                        WorkspaceSettings::get_global(cx).autosave,
-                        AutosaveSetting::OnWindowChange | AutosaveSetting::OnFocusChange
-                    ) {
-                        for item in pane.items() {
-                            Pane::autosave_item(item.as_ref(), self.project.clone(), cx)
+                    match WorkspaceSettings::get_global(cx).autosave {
+                        AutosaveSetting::OnWindowChange { trigger_formatter }
+                        | AutosaveSetting::OnFocusChange { trigger_formatter } => {
+                            for item in pane.items() {
+                                Pane::autosave_item(
+                                    item.as_ref(),
+                                    self.project.clone(),
+                                    trigger_formatter.unwrap_or(false),
```

Comment:
I'm wondering: can we get rid of that `unwrap_or(false)` here? I'm asking because I think ideally the default behavior is defined close to the type definition, not inside this code here.

Reply:
makes sense, will update this